Game/interface.py

```diff
@@ -168,6 +182,101 @@ def __init__(self, play_command, exit_command):
+                                    parent = self.frame)
+
+        music_label = DirectLabel(text = "Music",
+                                  pos = (-0.5, 0, 0.5),
+                                  scale = 0.1,
+                                  frameTexture = base.assets.sprite["frame"],
+                                  frameSize = (-3, 3, -0.5, 1),
+                                  parent = self.frame)
+
+        self.music_slider = DirectSlider(pos = (0.3, 0, 0.5),
+                                         scale = 0.4,
+                                         parent = self.frame,
+                                         value = self.music_volume,
+                                         command = self.update_music_volume)
+
+        sfx_label = DirectLabel(text = "SFX",
+                                pos = (-0.5, 0, 0.2),
+                                scale = 0.1,
+                                frameTexture = base.assets.sprite["frame"],
+                                frameSize = (-3, 3, -0.5, 1),
+                                parent = self.frame)
+
+        self.sfx_slider = DirectSlider(pos = (0.3, 0, 0.2),
+                                       scale = 0.4,
+                                       parent = self.frame,
+                                       value = self.sfx_volume,
+                                       command = self.update_sfx_volume)
+
+        ok_button = DirectButton(text = "Ok",
+                                 command = self.save_and_close,
+                                 pos = (-0.3, 0, -0.3),
+                                 scale = 0.1,
+                                 frameTexture = self.button_textures,
+                                 frameSize = (-3, 3, -0.5, 1),
+                                 clickSound = self.select_sfx,
+                                 rolloverSound = self.hover_sfx,
+                                 parent = self.frame)
+
+        back_button = DirectButton(text = "Back",
+                                    command = self.restore_and_close,
+                                    pos = (0.4, 0, -0.3),
+                                    scale = 0.1,
+                                    frameTexture = self.button_textures,
+                                    frameSize = (-3, 3, -0.5, 1),
+                                    clickSound = self.select_sfx,
+                                    rolloverSound = self.hover_sfx,
+                                    parent = self.frame)
+
+    def save_and_close(self):
+        '''Save options and return to parent menu.'''
+
+        log.info("Saving options...")
+        shared.MUSIC_VOLUME = self.music_volume
+        shared.SFX_VOLUME = self.sfx_volume
+
+        self.back_command()
+
+    def restore_and_close(self):
+        '''Restore previous options' values and return to parent menu.'''
+
+        base.musicManager.set_volume(self.old_music_volume)
```

Comment:
Use base.music_player.set_player_volume() instead (here and in other cases)

Reply:
Done